packages/component_code_gen/templates/generate_apps.py

```diff
@@ -0,0 +1,466 @@
+You should not return any text other than the code.
+
+output: Node.js code and ONLY Node.js code. You produce Pipedream component code and ONLY Pipedream component code. You MUST NOT include English before or after code, and MUST NOT include Markdown (like ```javascript) surrounding the code. I just want the code!
+
+## Pipedream Apps
+
+All Pipedream apps are Node.js modules that have a default export: an javascript object - a Pipedream app - as its single argument.
+It is essentially a wrapper on an API that requires authentication. Pipedream facades the authentication data in an object accessed by `this.$auth`. All app objects have three four keys: type, app, propDefinitions, and methods. The app object contains a `type` property, which is always set to "app". The `app` property is the name of the app, e.g. "google_sheets". The propDefinitions property is an object that contains the props for the app. The methods property is an object that contains the methods for the app.
+
+Here's an example Pipedream app for Raindrop:
+
+```javascript
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "app",
+  app: "raindrop",
+  propDefinitions: {
+    collectionId: {
+      type: "string",
+      label: "Collection ID",
+      description: "The collection ID",
+      async options() {
+        const { items } = await this.getCollections();
+        return items.map((e) => ({
+          value: e._id,
+          label: e.title,
+        }));
+      },
+    },
+    raindropId: {
+      type: "string",
+      label: "Bookmark ID",
+      description: "Existing Bookmark ID",
+      async options({
+        prevContext, collectionId,
+      }) {
+        const page = prevContext.page
+          ? prevContext.page
+          : 0;
+        const { items } = await this.getRaindrops(this, collectionId, {
+          page,
+        });
+        return {
+          options: items.map((e) => ({
+            value: e._id,
+            label: e.title,
+          })),
+          context: {
+            page: page + 1,
+          },
+        };
+      },
+    },
+  },
+  methods: {
+    async _makeRequest($ = this, opts) {
+      const {
+        method = "get",
+        path,
+        data,
+        params,
+        ...otherOpts
+      } = opts;
+      return axios($, {
+        ...otherOpts,
+        method,
+        url: `https://api.raindrop.io/rest/v1${path}`,
+        headers: {
+          ...opts.headers,
+          "user-agent": "@PipedreamHQ/pipedream v0.1",
+          "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        },
+        data,
+        params,
+      });
+    },
+    async getCollections($) {
+      return this._makeRequest($, {
+        path: "/collections",
+      });
+    },
+    async getRaindrops($, collectionId, params) {
+      return this._makeRequest($, {
+        path: `/raindrops/${collectionId}`,
+        params,
+      });
+    },
+  },
+};
+```
```

Comment:
Since we have 32K tokens, I'd also recommend passing a few more examples of full app files. We should be using as much of the 32k token limit as possible to try to get GPT to understand as much as possible about our API.